src/cargo_build.rs

```diff
@@ -586,3 +587,109 @@ impl CargoBuilder {
+
+        assert_eq!(actual_dir, expected_output_dir);
+        assert_eq!(actual_cargo_path, expected_cargo_path);
+        assert_eq!(cargo_builder.profile, expected_profile);
+        assert_eq!(cargo_builder.locked, true);
+    }
+
+    #[test]
+    fn construct_cargo_command() {
+        let mock = init_mock_env("/path_to_cargo", "debug", true);
+        let mut cargo_builder =
+            CargoBuilder::new(&mock, &Path::new("/path_to_output_directory"), false);
+        let cmd = cargo_builder.construct();
+        let actual_cmd = format!("{:?}", cmd);
+        let expected_cmd = match IS_RUST_STABLE_VERSION {
+            true => "\"/path_to_cargo\" \"build\" \"-vv\" \"--locked\"",
+            false => "cd \"/path_to_output_directory\" && \"/path_to_cargo\" \"build\" \"-vv\" \"--locked\"",
+        };
+
+        assert_eq!(actual_cmd, expected_cmd);
+    }
+
+    #[test]
+    fn change_cargo_path() {
+        let expected_cargo_path = "/path_to_cargo";
+        let expected_changed_cargo_path = "/changed/path_to_cargo";
+        let mock = init_mock_env(expected_cargo_path, "debug", true);
+        let mut cargo_builder =
+            CargoBuilder::new(&mock, &Path::new("/path_to_output_directory"), false);
+        let mut actual_cargo_path = cargo_builder.cargo_path.to_str().expect("Fail");
+
+        assert_eq!(actual_cargo_path, expected_cargo_path);
+
+        cargo_builder.cargo_path(Path::new(expected_changed_cargo_path));
+        actual_cargo_path = cargo_builder.cargo_path.to_str().expect("Fail");
+
+        assert_eq!(actual_cargo_path, expected_changed_cargo_path);
+    }
+
+    #[test]
+    fn change_target() {
+        let expected_target = "x86_64-unknown-linux-gnu";
+        let mock = init_mock_env("/path_to_cargo", "debug", true);
+        let mut cargo_builder =
+            CargoBuilder::new(&mock, &Path::new("/path_to_output_directory"), false);
+        let mut actual_target = cargo_builder.target.as_ref();
+
+        assert_eq!(actual_target, None);
+
+        cargo_builder.target(expected_target);
+        actual_target = cargo_builder.target.as_ref();
+
+        assert_eq!(actual_target.expect("Fail"), expected_target);
+    }
+
+    #[test]
+    fn change_rust_flags() {
+        let expected_rust_flags = ["-D", "warnings", "-C"];
+        let mock = init_mock_env("/path_to_cargo", "debug", true);
+        let mut cargo_builder =
+            CargoBuilder::new(&mock, &Path::new("/path_to_output_directory"), false);
+        let mut rust_flags: &Vec<String> = cargo_builder.rustflags.as_ref();
+
+        assert_eq!(rust_flags.is_empty(), true);
+
+        cargo_builder.add_rust_flags(&expected_rust_flags);
+        rust_flags = cargo_builder.rustflags.as_ref();
+
+        assert_eq!(rust_flags.len(), expected_rust_flags.len());
+        assert!(expected_rust_flags
+            .iter()
+            .all(|flag| rust_flags.contains(&flag.to_string())));
+    }
```

Comment:
:wrench: Suggest comparing as slices to have nicer test failure output
```suggestion
        assert_eq!(rust_flags.as_slice(), expected_rust_flags);
```

example failure output with slices
```console
thread 'cargo_build::tests::change_rust_flags' panicked at 'assertion failed: `(left == right)`
  left: `["-D", "warnings", "-C"]`,
 right: `["extra", "-D", "warnings", "-C"]`', src/cargo_build.rs:691:9
stack backtrace:
```
Versus 
```console
thread 'cargo_build::tests::change_rust_flags' panicked at 'assertion failed: `(left == right)`
  left: `3`,
 right: `4`', src/cargo_build.rs:691:9
 ```

Reply:
Nice! I'll change this